File: src/python_general_pkg/python_general_pkg/simple_subscriber_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleSubscriberNode(Node):
    def __init__(self, node_name):
        super().__init__(node_name)
        self._subscriber = self.create_subscription(
            msg_type = String, 
            topic = "/simple_topic", 
            callback = self._callback,
            qos_profile = 10, 
            )
        self.get_logger().info(f"订阅节点 {node_name} 已启动!")

    def _callback(self, msg):
        self.get_logger().info(f"订阅消息: {msg.data}")


def main():
    rclpy.init()
    node = SimpleSubscriberNode("python_simple_subscriber")
    rclpy.spin(node)
    rclpy.shutdown()

File: src/cpp_general_pkg/src/simple_subscriber_node.cpp
#include <string>

#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

using namespace std;
using std::placeholders::_1;


class SimpleSubscriberNode: public rclcpp::Node {

public:
    SimpleSubscriberNode(const string& node_name)
    : Node(node_name)
    {
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            "/simple_topic", 10, 
            std::bind(&SimpleSubscriberNode::callback, this, _1));

        RCLCPP_INFO(this->get_logger(), "订阅节点 cpp_simple_subscriber 已启动!");
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;

    void callback(const std_msgs::msg::String::SharedPtr msg) const
    {
        RCLCPP_INFO(this->get_logger(), "订阅消息: %s", msg->data.c_str());
    }
};


int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleSubscriberNode>("cpp_simple_subscriber");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
